pkg/k8s/pods/pod.go

```diff
@@ -76,19 +78,58 @@ func loopGetDevPod(dev *model.Dev, c *kubernetes.Clientset, waitUntilDeployed bo
 		return nil, err
 	}
 
-	log.Infof("pod %s with revison %s is progressing", pod.Name, d.Annotations[deploymentRevisionAnnotation])
-	if pod.Status.Phase == apiv1.PodRunning {
-		log.Infof("pod %s with revison %s is running", pod.Name, d.Annotations[deploymentRevisionAnnotation])
-		return pod, nil
+	opts := metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", OktetoInteractiveDevLabel, dev.Name),
+		FieldSelector: fmt.Sprintf("metadata.name=%s", pod.Name),
+	}
+
+	watchPod, err := c.CoreV1().Pods(dev.Namespace).Watch(opts)
```

Comment:
Do we get events even if they were created before the watcher? (e.g. what if the pod is already running by the time we run this code)